### PR TITLE
(#24) Set environment when managing resource

### PR DESCRIPTION
**Pull request by ZeroPointEnergy**
This makes it possible to manage resources like files with source parameters which retrieve files from a puppet master.

### CODE REVIEW

**Comment by ripienaar** [comment]
Thank you @ZeroPointEnergy this is interesting, please review [CONTRIBUTING.md] and we'll need those tests fixed - probably a simple fix for all failures!

File: agent/puppet.rb [hunk 1]
```diff
@@ -172,12 +172,18 @@ def run(command, options)
...
+        end
+
         if allow_managed_resources_management \
            || !@puppet_agent.managing_resource?(resource_name)
           resource = ::Puppet::Type.type(type).new(params)
           report = ::Puppet::Transaction::Report.new(:mcollective)
           ::Puppet::Util::Log.newdestination(report)
-          catalog = ::Puppet::Resource::Catalog.new
+          env = ::Puppet::Node::Environment.remote(environment)
+          catalog = ::Puppet::Resource::Catalog.new('choria', env)
```
Comment (ripienaar) [inline agent/puppet.rb]: hmm, I'm not familiar with these APIs so will have to have a closer look here

**Comment by ZeroPointEnergy** [comment]
Hi @ripienaar . Sorry if my PR was a bit premature. Will create a ticket, fix the tests and write some test for the feature.

**Comment by ZeroPointEnergy** [comment]
Tests are fixed now.

**Pull request merged**